torchaudio/backend/utils.py

```diff
@@ -1,62 +1,74 @@
+    no_backend,
+    sox_backend,
+    soundfile_backend,
+)
 
-_BACKEND = None
-_BACKENDS = {}
+__all__ = [
+    'list_audio_backends',
+    'get_audio_backend',
+    'set_audio_backend',
+]
 
 
-def list_audio_backends():
-    return list(_BACKENDS.keys())
+def list_audio_backends() -> List[str]:
+    """List available backends"""
+    backends = []
+    if is_module_available('soundfile'):
+        backends.append('soundfile')
+    if is_module_available('torchaudio._torchaudio'):
+        backends.append('sox')
+    return backends
 
 
-def set_audio_backend(backend: str) -> None:
-    """
-    Specifies the package used to load.
+def set_audio_backend(backend: Optional[str]) -> None:
+    """Set the backend for I/O operation
+
     Args:
         backend (str): Name of the backend. One of "sox" or "soundfile",
             based on availability of the system.
     """
-    if backend not in _BACKENDS:
+    if backend is not None and backend not in list_audio_backends():
         raise RuntimeError(
             f'Backend "{backend}" is not one of '
             f'available backends: {list_audio_backends()}.')
-    global _BACKEND
-    _BACKEND = backend
-
-
-def get_audio_backend() -> Optional[str]:
-    """
-    Gets the name of the package used to load.
-    """
-    return _BACKEND
 
+    if backend is None:
+        module = no_backend
+    elif backend == 'sox':
+        module = sox_backend
+    elif backend == 'soundfile':
+        module = soundfile_backend
+    else:
+        raise NotImplementedError(f'Unexpected backend "{backend}"')
 
-def _get_audio_backend_module() -> Any:
-    """
-    Gets the module backend to load.
-    """
-    if _BACKEND is None:
-        raise RuntimeError('Backend is not initialized.')
-    return _BACKENDS[_BACKEND]
+    for func in ['save', 'load', 'load_wav', 'info']:
+        setattr(torchaudio, func, getattr(module, func))
 
 
 def _init_audio_backend():
-    global _BACKEND
-    global _BACKENDS
-
-    _BACKENDS = {}
-    if _mod_utils.is_module_available('soundfile'):
-        _BACKENDS['soundfile'] = soundfile_backend
-    if _mod_utils.is_module_available('torchaudio._torchaudio'):
-        _BACKENDS['sox'] = sox_backend
-
-    if 'sox' in _BACKENDS:
-        _BACKEND = 'sox'
-    elif 'soundfile' in _BACKENDS:
-        _BACKEND = 'soundfile'
+    backends = list_audio_backends()
+    if 'sox' in backends:
+        set_audio_backend('sox')
+    elif 'soundfile' in backends:
+        set_audio_backend('soundfile')
     else:
         warnings.warn('No audio backend is available.')
-        _BACKEND = None
+        set_audio_backend(None)
+
+
+def get_audio_backend() -> Optional[str]:
+    """Get the name of the current backend"""
+    if torchaudio.load == no_backend.load:
+        return None
+    if torchaudio.load == sox_backend.load:
+        return 'sox'
+    if torchaudio.load == soundfile_backend.load:
+        return 'soundfile'
+    raise ValueError('Unknown backend.')
```

Comment:
nit: `f'Unexpected backend "{backend}"'` ?

Reply:
This state is not reachable and if it does that means user assigned something else to `torchaudio.load` and we have no idea what it is so, there is no `backend` variable we can get.